### ISSUE TITLE
Energy dashboard does not show gas consumption, why?

### ISSUE THREAD

**Issue by Surki69** [home-assistant/frontend#22337]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [ ] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

in HA energy dashboard i cannot get values from my Vitodens 300 and Vitovalor PA2 for gas consumption. The entities are available

### Describe the behavior you expected

graphs do not populate with gas consumption

### Steps to reproduce the issue

1. just followed the instructions in energy dashboard of HA and selected the correct entities from viessmann
2.
3.
...

### What version of Home Assistant Core has the issue?

2024.10.2

### In which browser are you experiencing the issue with?

safari 18.0.1

### Which operating system are you using to run this browser?

macOS 15.01

### State of relevant entities

```txt
sensor.e3_vitodens_300_0620_gasverbrauch_heizung_heute

restored: true
state_class: total_increasing
device_class: gas
friendly_name: Gasverbrauch Heizung Heute
supported_features: 0
unit_of_measurement: m³

sensor.e3_vitovalor_pa2_0419_gasverbrauch_heute

restored: true
state_class: total_increasing
device_class: gas
friendly_name: Gasverbrauch heute
supported_features: 0
unit_of_measurement: m³
```

### Additional information

not clear to me why it does not work if I choose the correct entities from the device. They show the values in device information with all the sensors but when I use these sensors in energy dashboard it doesn't work. I find this very odd.

**Comment by karwosts** [comment]
AFAIK `restored: true` implies that the entity is kind of abandoned and not provided by the integration anymore. 

Are you certain those are the same entities as you see in your device, they don't have a `_2` index appended to the end or anything like that?

**Comment by Surki69** [comment]
yes, these are the same entities. Actually the total gas counter for the vitodens starts reporting data but the vitavalor is still silent, don't know why.

**Comment by Surki69** [comment]
Does anyone know if I need two API keys if I run a vitodens and a vitovalor device? I am also not sure why Viessmann asks about IoT or User profile for an access token. What is the difference?

**Comment by Surki69** [comment]
It's working now. Thank you.
